src/librustc_typeck/check/mod.rs

```diff
@@ -1622,8 +1622,37 @@ pub fn check_item_type<'tcx>(tcx: TyCtxt<'tcx>, it: &'tcx hir::Item) {
                         ).emit();
                     }
 
-                    if let hir::ForeignItemKind::Fn(ref fn_decl, _, _) = item.kind {
-                        require_c_abi_if_c_variadic(tcx, fn_decl, m.abi, item.span);
+                    match &item.kind {
+                        hir::ForeignItemKind::Static(ty, _) => {
+                            struct HasTypeParameterVisitor { has_type_parameter: bool };
+
+                            impl<'tcx> Visitor<'tcx> for HasTypeParameterVisitor {
+                                fn nested_visit_map<'this>(
+                                    &'this mut self,
+                                ) -> NestedVisitorMap<'this, 'tcx> {
+                                    NestedVisitorMap::None
+                                }
+
+                                fn visit_path(&mut self, path: &hir::Path, _: hir::HirId) {
+                                    if let Res::Def(DefKind::TyParam, _) = path.res {
```

Comment:
Aside: I noticed that the following is allowed (no ICE, no error):
```rust
#![feature(const_generics)]

fn _f<const N: usize>() {
    extern "C" {
        static _a: [u8; N];
    }
}
```
cc @varkor

Reply:
@davidtwco: would it be possible to extend this to `HasNonLifetimeParameterVisitor` (or similar) and additionally check for `DefKind::ConstParam` (and add this test case)?

Reply:
@varkor I [discussed this with Centril](https://rust-lang.zulipchat.com/#narrow/stream/131828-t-compiler/topic/.2365133.20prohibit.20foreign.20statics/near/177478292) yesterday and decided that I'd aim to do this when I re-worked the PR to do this check for all statics in resolve.